data/pigui/modules/saveloadgame.lua

```diff
@@ -98,50 +99,40 @@ local function showSaveFiles()
 end
 
 local function closeAndClearCache()
-	ui.closeCurrentPopup()
-	ui.showSavedGameWindow = nil
+	ui.saveLoadWindow:close()
+	ui.saveLoadWindow.mode = nil
 	saveFileCache = {}
 	popupOpened = false
 end
 
 local function closeAndLoadOrSave()
 	if selectedSave ~= nil and selectedSave ~= '' then
-		if ui.showSavedGameWindow == "LOAD" then
+		if ui.saveLoadWindow.mode == "LOAD" then
 			Game.LoadGame(selectedSave)
-		elseif ui.showSavedGameWindow == "SAVE" then
+		elseif ui.saveLoadWindow.mode == "SAVE" then
 			Game.SaveGame(selectedSave)
 		end
 		closeAndClearCache()
 	end
 end
 
-local function savedGameWindow()
-	if ui.showSavedGameWindow then
-		if not popupOpened then
-			popupOpened = true
-			ui.openPopup("LoadGame")
-		end
+ui.saveLoadWindow = ModalWindow.New("LoadGame", function()
+	local mode
+	mode = ui.saveLoadWindow.mode == 'SAVE' and lui.SAVE or lui.LOAD
```

Comment:
Is there a reason why you're declaring `mode` and then assigning a line later? This should be all one line.

Reply:
I think that's still from the pre-modal save window days, I just didn't touch it when I made the window modal. Fixed now.